### PR TITLE
fix: Fix tooltip distance

### LINKED ISSUE #23859: [Bug]: Permissions DQA
### Describe the bug

- [x] Navigating from Connected dApp: https://github.com/MetaMask/metamask-extension/pull/24085
Currently when clicking the back arrow a user navigates back to the Wallet screen. This should probably take the user to the All Permissions screen instead.

https://github.com/MetaMask/metamask-extension/assets/152526076/b8a3cd4c-daf5-46ee-82df-2b77cc7e933b

- [x] Disable Connect more accounts button: https://github.com/MetaMask/metamask-extension/pull/24084
When there are no accounts to connect the Connect more accounts button on the screen should be disabled rather than showing an empty modal

https://github.com/MetaMask/metamask-extension/assets/152526076/6fb50964-c1f1-4e3a-97fd-10062a769191

- [x] Permissions tooltip within Connect Accounts Modal: https://github.com/MetaMask/metamask-extension/pull/24098
The permissions tooltip is not displaying correctly within the connect accounts modal. Additionally there should be a 4px spacing between the tooltip info icon and the permissions text label.

Current:
image

Desired:
image

- [ ] Connected site Section titles
If a site is not connected to a snap, there is no need to show a Connected accounts section title

image

- [ ] All permissions missing tabs
Snaps and sites should be separated by tabs rather than section titles.

Current:
image

Desired:
image

- [x] On-hover behavior for Connected Site on All Permissions screen: https://github.com/MetaMask/metamask-extension/pull/24099
Sites listed on the All Permissions screen should have an [on-hover bg color](https://metamask.github.io/metamask-storybook/?path=/docs/components-ui-listitem--docs) to indicate that they are clickable and will take users to a different screen.

https://github.com/MetaMask/metamask-extension/assets/152526076/e98d2ad2-d948-4aab-8982-aa742af769ce

- [ ] Snap list item on All Permissions screen
Snaps listed on the All Permissions screen dont match the design. 

Current:
image

Desired:
image

### Expected behavior

_No response_

### Screenshots/Recordings

_No response_

### Steps to reproduce

...

### Error messages or log output

_No response_

### Version

11.13.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by jonybur**
## **Description**

The permissions tooltip is not displaying correctly within the connect accounts modal. Additionally there should be a 4px spacing between the tooltip info icon and the permissions text label.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24098?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/23859

## **Manual testing steps**

1. Go to this page...
2.
3.

### **Before**

Screenshot 2024-04-18 at 03 21 07

### **After**

Screenshot 2024-04-18 at 03 13 02

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/connect-accounts-modal/connect-accounts-modal-list.tsx [hunk 1]
```diff
@@ -71,11 +71,17 @@ export const ConnectAccountsModalList: React.FC<ConnectAccountsListProps> = ({
               display={Display.Flex}
             >
               <Tooltip
+                distance={75}
```
Comment (NidhiKJha) [inline ui/components/multichain/connect-accounts-modal/connect-accounts-modal-list.tsx:74]: Why do we need this property?

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
Good start! Position prop fix the tooltip positioning issue and we don't need the distance prop. 
Screenshot 2024-04-18 at 2 58 12 PM

**Comment by jonybur** [comment]
@NidhiKJha 

With the marginInlineEnd attribute:

Screenshot 2024-04-18 at 19 01 48

With the distance attribute:

Screenshot 2024-04-18 at 19 17 16

The distance attribute, according to React Tippy, is `how far in pixels the tippy element is from the reference element.`

**Comment by NidhiKJha** [comment]
> Fix distance, marginInlineEnd

@jonybur Marginline is fine. But I still don't think we need distance here.` position="top"` prop is handling the position of tooltip

**Comment by jonybur** [comment]
@NidhiKJha It is handling the position but not the distance to the icon. The bug shows a desired positioning where the tooltip is further away from the icon than what we currently render.

This is the desired design:

Screenshot 2024-04-19 at 11 45 42

Without the `distance` attribute, the tooltip renders too close to the (i) icon

**Review by NidhiKJha — APPROVED** [review]

**Comment by EtherWizard33** [comment]
not sure about the positioning on the left of the tooltip in the 'After' picture above, https://github.com/MetaMask/metamask-extension/pull/24098#:~:text=Before-,After,-Pre%2Dmerge%20author

seems like its hugging the left side, not sure if thats intended already talked about it with Jony

**Review by darkwing — APPROVED** [review]

**Pull request merged**